src/modules/site/base/views/mapping.py

```diff
@@ -89,7 +128,7 @@ def mapping_report(id):
   fluid_container = True
   data_url = generate_blob_url(mapping.get_bucket_name(), mapping.get_data_blob_path())
   # if hasattr(mapping, 'mapping_report_url'):
-  if hasattr(mapping, 'report_path'):
+  if mapping.report_path is not None:

```

Comment:
careful -- mapping might be invalid if the <id> was invalid or didn't match a correct mapping.

Reply:
thanks for the nuanced response! My original comment was rendered incorrectly, it was missing the `<id>` bit, sry about that if made the comment unclear. The main concern was about `mapping` being `None`, which has been covered. 

Regarding your question about the failed calculation -- if the pipeline errors out, there are two levels of logs that we capture -- the "platform error" which happens before the nextflow pipelines start, and then the errors generated by nextflow. We can look at how these are displayed in other tasks and unify them across tools. Let's discuss.

Reply:
(Submission being `None`, as well as some of my concerns above, fixed in 0f531415cf290a2cc0ed24dd25e74b47b132c058)